File: sorting/selectionSort.js
const selectionSort = (arr) => {
    let minIndex = 0
    for(let i = 0; i < arr.length; i++) {

        minIndex = i

        for(let j = i + 1; j < arr.length; j++) {
            if(arr[j] < arr[minIndex]) {
                minIndex = j
            }
        }

        let temp = arr[i]
        arr[i] = arr[minIndex]
        arr[minIndex] = temp
    }
    return arr
};

console.log(selectionSort([64, 25, 12, 22, 11]))
console.log(selectionSort([6, 5, 4, 3, 2, 1]))

File: sorting/selectionSort.cpp
#include <iostream>
#include <cmath>
#include <climits>

int* selectionSort(int[], int);

int main() {
    const int SIZE = 7;
    int a[SIZE] = {6, 2, 3, 8, 9, 10, 1};

    selectionSort(a, SIZE);

    for(int i = 0; i < SIZE; i++)
        std::cout << a[i] << " ";
    std::cout << std::endl;

    return 0;
}

int* selectionSort(int a[], int size) {
    for(int i = 0; i < size; i++) {
        int index = i;
        for(int j = i + 1; j < size; j++) {
            if(a[j] <= a[index]) {
                index = j;
            }
        }

        int temp = a[i];
        a[i] = a[index];
        a[index] = temp;
    }
    return a;
}